### ISSUE TITLE
Minimap

### ISSUE THREAD

**Issue by JosephTLyons** [zed-industries/feedback#263]

**Comment by exalted** [comment]
FWIW, I am strongly against this 👎

**Comment by aexvir** [comment]
I'm also not a fan, it brings me zero value and it's distracting and takes space
but as long as it's optional I don't see why it couldn't be added
there is definitely people who likes it

**Comment by aexvir** [comment]
can't see anything 😅 did the screenshot get attached?

**Comment by scott-ling** [comment]
never said I was intelligent, seems github email drops images

![image](https://user-images.githubusercontent.com/82799158/179071394-81ec65e5-9ea5-4eef-b827-8cad51b218b0.png)

**Comment by aexvir** [comment]
oh, hold up, that's something different... that's the scrollbar itself, which I have butchered with custom css 🤣 
yep, this is what I know from intellij based editors, but I wasn't aware vscode also had it
I turned off scrollbars looong time ago 😅 

yeah, this is definitely more useful 👍🏻

**Comment by scott-ling** [comment]
yeah, minimaps sux, minimap like features in scroll bar, awesome use of space

**Comment by JosephTLyons** [comment]
I'm dyslexic (being a dyslexic dev is a weird thing, but that's another story), so I often recognize code by the shape of it - which is where the minimap becomes useful for me.

**Comment by uicrafts** [comment]
I think the minimap by itself is not very useful. However, it's actually really nice when it shows red/yellow lines for ESLint and/or TypeScript errors/warnings throughout the entire file, allowing you to quickly click on that area of the minimap and scrolling you down to where the error is.

**Comment by heyallan** [comment]
I also recognize code by it's shape. It's super useful. But I prefer seeing an outline of the document. It gives me shape but also meaning in less space. Like a table of contents.

The breadcrumb feature already creates an outline, but only line by line, which is weird because if I am already at that line I already know where I am.

Having a full vertical outline would be useful for recognizing large documents.

Also, hints in the scrollbar are immediately useful.

**Comment by JosephTLyons** [comment]
@heyallan,

You're probably already aware of this, but in case you haven't seen this, we do have a `search buffer symbols` action that gives you a modal popup where you can see the overview of the current file's symbols.  You can then filter down to what you want and be taken to that point in the code.

<img width="854" alt="SCR-20221220-lup" src="https://user-images.githubusercontent.com/19867440/208762824-32f104d7-3976-4482-be59-8535bd6fd545.png">

**Comment by heyallan** [comment]
@JosephTLyons Exactly. The functionality is already there. I just would like to have it permanently open on the side as a form of minimap.

**Comment by natempv** [comment]
fwiw I love the minimap for giving the code a shape and body, helps me navigate it by feel, like a few people have mentioned already. like i know where things are from the topology of the thing, like it's a kaiju or a battleship. i've come to understand this is atypical but

also because I'll occasionally annotate my code with huge ascii/ansi art, kind of like xcode's //MARK comments but I usually just use figlet

**Comment by wzxu** [comment]
Also using the minimap to quickly navigate code by shape and colors. 
Would love a similar option to not render actual characters (makes the shape & color fuller and easier to scan).

<img width="462" alt="image" src="https://user-images.githubusercontent.com/5619348/229171672-b343527a-df8f-4c1f-abda-0e3c20f5b88a.png">

![image](https://user-images.githubusercontent.com/5619348/229174027-bf1d7775-d89d-4887-a2bd-3240e6dc0214.png)

**Comment by ojkelly** [comment]
I've found minimaps really useful for many of the reasons mentioned here. While I am missing them in Zed, solving for that might not require a minimap.

First, I rebound the outline to `cmd-o` to make it much quicker to open. The fact it's syntax highlighted, and searchable makes it _more useful than a minimap for moving around_ in a file.

```json
  {
    "bindings": {
      "cmd-o": "outline::Toggle"
    }
  }
```

However, I still find myself missing how vscode indicates problems in the file tree and minimap.

The problem multi-buffer is a really neat idea, but I don't find it useful for more complex problems. (The amount of padding and UI chrome can sometimes make things feel disconnected).

I think I could live without a minimap if the file tree and outline had a color/underline/dot/something to indicate a problem/warning.

I seem to be a very spatial programming, in that I rarely search, and instead navigate to the file and part of the file when looking for something, or solving a bug. I think this is my brain indexing on where something is more than what it was named.

Given Zed's opportunity to rethink much of the IDE, and as much as I do love minimaps, I'm keen to see what you folks come up with.

**Comment by ShayTSC** [comment]
> @heyallan,
> 
> You're probably already aware of this, but in case you haven't seen this, we do have a `search buffer symbols` action that gives you a modal popup where you can see the overview of the current file's symbols. You can then filter down to what you want and be taken to that point in the code.
> 
> <img alt="SCR-20221220-lup" width="854" src="https://user-images.githubusercontent.com/19867440/208762824-32f104d7-3976-4482-be59-8535bd6fd545.png">

This feature is really good on backend, but for the frontend code sometimes you have some annoying and long JSX function, in symbol view it looks like this. (Not very helpful) It will be so good if that symbol view can parse JSX!

<img width="842" alt="CleanShot 2023-04-07 at 13 45 19@2x" src="https://user-images.githubusercontent.com/10642396/230549001-d16fea41-c360-4086-84a6-4c0b3bb58c56.png">

**Comment by exalted** [comment]
> I seem to be a very spatial programming, in that I rarely search, and instead navigate to the file and part of the file when looking for something, or solving a bug. I think this is my brain indexing on where something is more than what it was named.

I love this insight!

**Comment by cktang88** [comment]
I think it would be nice to have configurable minimap, either in the scrollbar or standalone, since there's no clear consensus here.

**Comment by arguiot** [comment]
I really got used to minimaps. I'm not dyslexic but I do use it to recognise the shape of the code on long files. The other use case that I found SUPER useful is the `MARK` statements in Xcode.

**Comment by aeturnum** [comment]
Hey there - very excited about Zed - I wanted to echo others' comments about coming to rely on minimaps to navigate larger projects. At this point the lack of the option is a blocker on using adopting any new editor as a main editor.

I understand why people bring up `search buffer symbols` but I want to emphasize how the minimap feature allows you to spatially organize files by section and recognize layout. I both understand why some people don't care and want to encourage the Zed team to think about this as a core feature of a modern editor.

**Comment by fxsalazar** [comment]
Here's the alternative [Display hunks/diagnostics/search results in scroll bar](https://github.com/zed-industries/community/issues/1305) that I'm personally looking forward to... Let's 👍🏽  it up.

**Comment by JosephTLyons** [comment]
> Here's the alternative [Display hunks/diagnostics/search results in scroll bar](https://github.com/zed-industries/community/issues/1305) that I'm personally looking forward to and kind of mentioned here https://github.com/zed-industries/community/issues/263#issuecomment-1287994721
> 
> Let's 👍🏽  it up.

Git diff in scrollbar will be landing this next Wednesday, if all goes to plan.

**Comment by restfuladi** [comment]
Any chance of ticks in the scrollbar for bookmarks/errors/warnings/todo/etc like JetBrains? Even better if they were clickable.

**Comment by Congyuwang** [comment]
Xcode has one of the most useful minimap I've ever used:

You can use `// MARK: - XXX` to add section header, which will be displayed in the minimap, making it super easy to navigate around:
<img width="988" alt="截圖 2023-09-06 13 53 58" src="https://github.com/zed-industries/community/assets/52687642/18480c64-10b8-4e3e-b42b-7aec8e381409">

Putting the mouse over the minimap shows the name of the function the user is pointing at:
<img width="477" alt="截圖 2023-09-06 13 56 13" src="https://github.com/zed-industries/community/assets/52687642/bc1b5aa4-ac36-4126-9a03-21e713c9fe44">

**Comment by aranw** [comment]
I think one of the most useful features of the minimap that I'm missing in Zed is the ability to quickly see where symbols are being used 

![Screenshot 2023-09-26 at 09 27 38](https://github.com/zed-industries/community/assets/241565/c8136d23-150b-4fc5-a7bf-ff671cd31e84)

**Comment by fxsalazar** [comment]
Is not only about the theme, but having it in the scrollbar too...
